src/text_to_binary/rewrite.ml

```diff
@@ -143,32 +132,50 @@ let rewrite_expr (modul : Assigned.t) (locals : binary param list)
       let id = find_local id in
       Ok (Local_tee id)
     | If_else (id, bt, e1, e2) ->
-      let* bt = bt_to_raw bt in
+      let* bt =
+        match bt with
+        | Some bt ->
+          let+ bt = rewrite_block_type (typemap modul.typ) modul bt in
+          Some bt
+        | None -> Ok None
+      in
       let block_ids = id :: block_ids in
       let* e1 = expr e1 (loop_count, block_ids) in
       let+ e2 = expr e2 (loop_count, block_ids) in
       If_else (id, bt, e1, e2)
     | Loop (id, bt, e) ->
-      let* bt = bt_to_raw bt in
+      let* bt =
+        match bt with
+        | Some bt ->
+          let+ bt = rewrite_block_type (typemap modul.typ) modul bt in
+          Some bt
+        | None -> Ok None
+      in
       let+ e = expr e (loop_count + 1, id :: block_ids) in
       Loop (id, bt, e)
     | Block (id, bt, e) ->
-      let* bt = bt_to_raw bt in
+      let* bt =
+        match bt with
+        | Some bt ->
+          let+ bt = rewrite_block_type (typemap modul.typ) modul bt in
+          Some bt
+        | None -> Ok None
+      in
```

Comment:
This pattern is repeated many times, could you make a helper function to make it shorter?